File: fdlsandbox-android-java/app/src/main/java/com/omatt/fdlsandbox/modules/deeplink/DeepLinkContract.java
package com.omatt.fdlsandbox.modules.deeplink;

import android.content.Intent;

import com.omatt.fdlsandbox.BasePresenter;
import com.omatt.fdlsandbox.BaseView;

/**
 * Created by omarmatthew on 9/11/17.
 */

public interface DeepLinkContract {
    interface View extends BaseView<Presenter> {
        void updateDeepLink(String deepLink);

        void updateInviteId(String inviteId);

        void updateReferrerId(String referrerId);
    }
    interface Presenter extends BasePresenter<View> {
        void processDeepLink(Intent intent);
    }
}

File: fdlsandbox-android-kotlin/app/src/main/java/com/omatt/fdlsandbox/modules/deeplink/DeepLinkContract.kt
package com.omatt.fdlsandbox.modules.deeplink

import android.content.Intent
import com.omatt.fdlsandbox.BasePresenter
import com.omatt.fdlsandbox.BaseView

/**
 * Created by omarmatthew on 9/16/2017.
 */
interface DeepLinkContract {
    interface View : BaseView<Presenter> {
        fun updateDeepLink(deepLink: String)

        fun updateInviteId(inviteId: String)

        fun updateReferrerId(referrerId: String)
    }

    interface Presenter : BasePresenter<View> {
        fun processDeepLink(intent: Intent)
    }
}